### PR TITLE
HHH-17187 - Disallow UUIDs with a trailing 0 byte in tests

### PR DESCRIPTION
**Pull request by jrenaat**
https://hibernate.atlassian.net/browse/HHH-17187

### CODE REVIEW

File: hibernate-core/src/main/java/org/hibernate/id/uuid/SafeRandomUUIDGenerator.java
Comment (beikov) [inline hibernate-core/src/main/java/org/hibernate/id/uuid/SafeRandomUUIDGenerator.java:1]: I don't think this belongs into the src. This should only be part of e.g. hibernate-testing. I don't know if the "real" Sybase ASE also trims trailing zero bytes or if this is just a bug in our version, but I would assume the latest patch version has this fixed.

File: tooling/metamodel-generator/src/main/java/org/hibernate/jpamodelgen/validation/MockSessionFactory.java [hunk 1]
```diff
@@ -177,7 +177,7 @@ public Class<?> classForName(String className) {
 		metamodel = new MockMappingMetamodelImpl();
 
 		bootModel = new MetadataImpl(
-				UUID.randomUUID(),
+				SafeRandomUUIDGenerator.safeRandomUUID(),
```
Comment (beikov) [inline tooling/metamodel-generator/src/main/java/org/hibernate/jpamodelgen/validation/MockSessionFactory.java:180]: This change seems unnecessary.

**Review by beikov — COMMENTED** [review]
The changes look fine to me, but I think we also have some tests that use our builtin uuid generators, for which you'd need to replace/configure the `StandardRandomStrategy`.

File: hibernate-core/src/main/java/org/hibernate/id/uuid/SafeRandomUUIDGenerator.java
Comment (jrenaat) [reply hibernate-core/src/main/java/org/hibernate/id/uuid/SafeRandomUUIDGenerator.java:1]: > I don't think this belongs into the src. This should only be part of e.g. hibernate-testing.
Funny you would say that, I actually started out in hibernate testing, and then moved it to core ... I'll move it back.

> I don't know if the "real" Sybase ASE also trims trailing zero bytes or if this is just a bug in our version, but I would assume the latest patch version has this fixed.
Meaning what? That this work is useless now?

File: tooling/metamodel-generator/src/main/java/org/hibernate/jpamodelgen/validation/MockSessionFactory.java (hunk 1 above)
Comment (jrenaat) [reply tooling/metamodel-generator/src/main/java/org/hibernate/jpamodelgen/validation/MockSessionFactory.java:180]: Not sure if it's absolutely necessary, but is it a problem?

File: hibernate-core/src/main/java/org/hibernate/id/uuid/SafeRandomUUIDGenerator.java
Comment (beikov) [reply hibernate-core/src/main/java/org/hibernate/id/uuid/SafeRandomUUIDGenerator.java:1]: No no, this is just a justification for not putting this work into the source of hibernate-core, but rather move it to hibernate-testing.

File: tooling/metamodel-generator/src/main/java/org/hibernate/jpamodelgen/validation/MockSessionFactory.java (hunk 1 above)
Comment (beikov) [reply tooling/metamodel-generator/src/main/java/org/hibernate/jpamodelgen/validation/MockSessionFactory.java:180]: Let's not change code that doesn't have to change. Please revert this one change.

**Comment by jrenaat** [comment]
> The changes look fine to me, but I think we also have some tests that use our builtin uuid generators, for which you'd need to replace/configure the `StandardRandomStrategy`.

I've only found a couple of uses of `@UuidGenerator`, and `CustomVersionOneStrategy` and and those are testing the correct working of those, so perhaps we shouldn't replace them?

**Comment by beikov** [comment]
It's not just the `@UuidGenerator` that makes use of the `StandardRandomStrategy`. Every entity that uses `@GeneratedValue` on an UUID typed attribute will use that well. You can use the following regex pattern to find some uses `@GeneratedValue[^@]+UUID`, but I suppose it would be easiest to try running the whole testsuite in IntelliJ in debug mode, setting a break point in `StandardRandomStrategy` and just trace back to where this comes from.

**Comment by jrenaat** [comment]
> It's not just the `@UuidGenerator` that makes use of the `StandardRandomStrategy`. Every entity that uses `@GeneratedValue` on an UUID typed attribute will use that well. You can use the following regex pattern to find some uses `@GeneratedValue[^@]+UUID`, but I suppose it would be easiest to try running the whole testsuite in IntelliJ in debug mode, setting a break point in `StandardRandomStrategy` and just trace back to where this comes from.

Ok.
Tbh I've also been staring myself blind on how to replace the `StandardRandomStrategy` with another one ... at least in the new `UuidGenerator`this doesn't seem to be possible ... ?

**Comment by beikov** [comment]
>Tbh I've also been staring myself blind on how to replace the `StandardRandomStrategy` with another one ... at least in the new `UuidGenerator`this doesn't seem to be possible ... ?

True, so I think it might be best to try disable tests that use that for Sybase, possibly reduce the amount of tests that use that to avoid skipping to many tests for Sybase. By reducing the amount I mean that we should switch to a different id type for the entity of that test class, unless the test is explicitly about testing UUID.

**Pull request merged**